lib/sbi_trap.c

```diff
@@ -178,6 +179,8 @@ void sbi_trap_handler(struct sbi_trap_regs *regs,
 		msg = "ecall handler failed";
 		break;
 	default:
+		/* If the trap came from S or U mode, redirect it there */
+		rc = sbi_trap_redirect(regs, scratch, regs->mepc, mcause, mtval);
```

Comment:
This sbi_trap_redirect() call should under

```
if (((regs->mstatus & MSTATUS_MPP) >> MSTATUS_MPP_SHIFT) != PRV_M) {
...
}

```

Reply:
From sbi_trap_redirect:

/* Sanity check on previous mode */
	prev_mode = (regs->mstatus & MSTATUS_MPP) >> MSTATUS_MPP_SHIFT;
	if (prev_mode != PRV_S && prev_mode != PRV_U)
return SBI_ENOTSUPP;

Reply:
Ahh yes, I forgot. We are already checking this in sbi_trap_redirect.